File: Java/src/pl/eco/roslina/Guarana.java
package pl.eco.roslina;

import pl.eco.Organizm;
import pl.eco.Roslina;
import pl.eco.Swiat;

public class Guarana extends Roslina {
    public Guarana(int x, int y, Swiat moj) {
        sila = 0;
        inicjatywa = 0;
        nazwa = "guarana";
        symbol = 'g';
        wiek = 0;
        this.x = x;
        this.y = y;
        this.moj = moj;
    }

    public Guarana(int x, int y,Swiat w1,int wiek){
        sila = 0;
        inicjatywa = 0;
        nazwa = "guarana";
        symbol = 'g';

        this.wiek = wiek;
        this.x = x;
        this.y = y;
        this.moj = moj;
    }

    @Override
    public boolean obrona(Organizm atujacy) {
        atujacy.setSila(atujacy.getSila() + 3);
        this.czyZyje=false;
        atujacy.moj.komentarze.add("Guarana wzmocnila "+atujacy.getNazwa());
        return false;
    }
}

File: C++/Symulator2/Guarana.cpp
#include "Guarana.h"
#include <iostream>
#include <Windows.h>

using namespace std;

Guarana::Guarana(int x, int y, Swiat* moj) {
	sila = 0;
	inicjatywa = 0;
	nazwa = "guarana";
	symbol = 'g';
	wiek = 0;
	this->x = x;
	this->y = y;
	this->moj = moj;
}

bool Guarana::obrona(Organizm* atujacy) {
	cout << "Guarana zjedzona przez " << atujacy->getNazwa() << " jego sila wzrasta o 3" << endl;
	Sleep(2000);
	atujacy->SetSila(atujacy->getSila() + 3);
	return false;
}